### DISCUSSION TITLE
关于ApiObject几点不懂的问题

### DISCUSSION THREAD

**Discussion by Kbstsn** [XTLS/Xray-core#2904] [Q&A]
```javascript
{
    "api": {
        "tag": "api",
        "server": [
            "StatsService"
        ]
    },
    "inbounds": [
        {
            "listen": "127.0.0.1",
            "port": 10085,
            "protocol": "dokodemo-door",
            "settings": {
                "address": "127.0.0.1"
            },
            "tag": "api"
        }
    ],
    "routing": {
        "rules": [
            {
                "inboundTag": [
                    "api"
                ],
                "outboundTag": "api",
                "type": "field"
            }
        ]
    }
}
``` 
我看文档给的示例是这样写的，但是并没有说需要一个出站流量先指向127.0.0.1:10085端口那么它是怎么获取到流量的呢？还有就是路由规则里inboundTag和outboundTag都是api，那么流量又是怎么出去的呢？另外就是ApiObject里的api的tag作用是什么？

**Comment by vrnobody** [comment]
core遇到`{“api":{}}`配置项的时候会生成一个tag相同的outbound。因为routing里面指定了"api"->"api"，所以dokodemo-door(10085)端口的流量会转发到这个outbound里面。数据格式是protobuf。这些数据会传递给core内部的grpc服务器处理。
所以如果你开启了`handlerservice`并执行`./xray api rmo --server=127.0.0.1:10085 "api"`，那么api功能就会失效。

我只知道这么多，再深入点的原理我也不是很清楚。
